### PR TITLE
Create Separate Pages for Icon Examples

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
The terra-icon site page was very long and difficult to use. This PR breaks out the static and themeable icon examples into their own sub-pages.

Addresses #447.

### CODE REVIEW

File: packages/terra-site/src/examples/icon/components/IconThemeable.jsx [hunk 1]
```diff
@@ -0,0 +1,1134 @@
...
+import IconCompose from 'terra-icon/lib/icon/IconCompose';
+import IconUnread from 'terra-icon/lib/icon/IconUnread';
+import IconSwap from 'terra-icon/lib/icon/IconSwap';
+import IconMicrophone from 'terra-icon/lib/icon/IconMicrophone';
+import IconMicrophoneDisabled from 'terra-icon/lib/icon/IconMicrophoneDisabled';
+import IconOutgoingCall from 'terra-icon/lib/icon/IconOutgoingCall';
+import IconIncomingCall from 'terra-icon/lib/icon/IconIncomingCall';
+import IconMissedCall from 'terra-icon/lib/icon/IconMissedCall';
+const IconAll = () => (
+  <div>
```
Comment (emilyrohrbough) [inline packages/terra-site/src/examples/icon/components/IconThemeable.jsx]: The change in this file includes adding the version, the Themeable Icons heading and fixing indentation.

File: packages/terra-site/src/examples/icon/IconStatic.jsx [hunk 2]
```diff
@@ -1,241 +0,0 @@
-// This file is autogenerated from scripts/src/generate-example
```
Comment (emilyrohrbough) [inline packages/terra-site/src/examples/icon/IconStatic.jsx]: Moved to components folder

File: packages/terra-site/src/examples/icon/IconThemeable.jsx [hunk 3]
```diff
@@ -1,1129 +0,0 @@
-// This file is autogenerated from scripts/src/generate-example
```
Comment (emilyrohrbough) [inline packages/terra-site/src/examples/icon/IconThemeable.jsx]: Moved to components folder

**Comment by mmalaker** [comment]
These look good, but I'd suggest adding some explanatory text to each sub-page. Something like, "The static icons are designed to convey a specific meaning. As such they are to be used without modification." and "The themeable icons may be modified to match specific application styles."

For reference, we should link to the Theme Provider page as well:
https://terra-ui.herokuapp.com/components/core/theme-provider

**Comment by emilyrohrbough** [comment]
@mmalaker I've updated the icon subpages. Currently there are not themes available for Icons which is why I didn't reference the theme provided.

**Comment by mmalaker** [comment]
just spoke to @neilpfeiffer. Let's add the links to the icon sub-pages from the main page. They can be added under the "Usage" section, like we have for forms.

**Comment by emilyrohrbough** [comment]
@mmalaker The links provided in the Form's main page are links to each sub-component's README page on the github repo. If you look at these icon sub-pages, they both provide information regarding the icon, icon concept and icon usage and this is the same information the form's subcomponent Readme provides.

**Comment by mmalaker** [comment]
My mistake. @neilpfeiffer, do you still want the links on the main page or are we good without them for now?

**Comment by neilpfeiffer** [comment]
Talked with @emilyrohrbough and we came up with some ideas of how to enhance the page template generally, so we'll gather our thoughts and put into a separate issue.  

We can get by without the child links for now and get them added later.

**Pull request merged**